src/main/java/com/google/cloud/genomics/dataflow/functions/KeyReadsFn.java

```diff
@@ -52,15 +60,31 @@ public KeyReadsFn() {
   
   @Override
   public void startBundle(Context c) {
-  lociPerShard = c.getPipelineOptions()
+    lociPerShard = c.getPipelineOptions()
       .as(Options.class)
       .getLociPerWritingShard();
+    lastPos = -1;
+    count = 0;
+  }
+  
+  @Override
+  public void finishBundle(Context c) {
+    LOG.info("KeyReadsDone: Processed " + count + " reads" + "min=" + minPos + 
+        " max=" + maxPos);
   }
 
   @Override
   public void processElement(DoFn<Read, KV<Contig, Read>>.ProcessContext c)
     throws Exception {
     final Read read = c.element();
+    long pos = read.getAlignment().getPosition().getPosition();
+    if (lastPos > pos) {
+      LOG.info("Out of order read at pos "+ lastPos + " " + pos);
```

Comment:
This is a leftover form the debugging I was doing - I wanted to confirm that when the data is read from the API via gRPC it does indeed come in coordinate order.
A bunch of optimizations rely on the fact that this is indeed the case, otherwise we have to reshard/reorder reads which is costly, especially in the case of full genome export (we end up materializing the whole thing several times in the process).
In the current implementation we in fact do not need to key the reads because we do not need to reshard them - we rely on reading side sharding and do not re-shard again for writing.
For the purposes of this function if it is used in other scenarios, where we do need resharding, the ordering does not strictly matter (as Paul says), so I removed the logging.
If some future implementation wants to rely on the ordering within a bundle we could implement a varian of KeyReadsFn that validates ordering and fails if it observes an out of order read but for now there is no concrete need for this.